lib/crontab/cron_expression/parser.ex

```diff
@@ -193,28 +201,54 @@ defmodule Crontab.CronExpression.Parser do
   defp tokenize(interval, :complex_divider, value) do
     [base, divider] = String.split(value, "/")
 
-    case {tokenize(interval, base), Integer.parse(divider, 10)} do
-      {{:ok, clean_base}, {clean_divider, _}} ->
+    # Range increments apply only to * or ranges in <start>-<end> format
+    range_tokenization_result = tokenize(interval, :-, base)
+    other_tokenization_result = tokenize(interval, base)
+    integer_divider = Integer.parse(divider, 10)
+
+    case {range_tokenization_result, other_tokenization_result, integer_divider} do
+      # Found range in <start>-<end> format
+      {{:ok, clean_base}, _, {clean_divider, _}} ->
         {:ok, {:/, clean_base, clean_divider}}
 
-      {_, :error} ->
-        {:error, "Can't parse " <> value <> " as interval " <> Atom.to_string(interval) <> "."}
+      # Found star (*) range
+      {{:error, _}, {:ok, :*}, {clean_divider, _}} ->
+        {:ok, {:/, :*, clean_divider}}
 
-      {error = {:error, _}, _} ->
+      # No valid range found
+      {error = {:error, _}, _, _} ->
         error
     end
   end
 
   @spec clean_value(CronExpression.interval(), binary) ::
           {:ok, CronExpression.value()} | {:error, binary}
+
+  defp clean_value(:minute, value) do
+    clean_integer_within_range(value, "minute", @minute_values)
+  end
+
+  defp clean_value(:hour, value) do
+    clean_integer_within_range(value, "hour", @hour_values)
+  end
+
   defp clean_value(:weekday, "L"), do: {:ok, 7}
 
   defp clean_value(:weekday, value) do
```

Comment:
Can we split this function body to avoid a too long function and deep nesting?

```
$ mix credo --strict
Checking 7 source files ...

info: the following checks were skipped because they're not compatible with
your version of Elixir (1.8.0).

   1) Credo.Check.Refactor.MapInto
   2) Credo.Check.Warning.LazyLogging


  Refactoring opportunities
┃
┃ [F] → Function is too complex (CC is 14, max is 9).
┃       lib/crontab/cron_expression/parser.ex:237:8 #(Crontab.CronExpression.Parser.clean_value)
┃ [F] → Function body is nested too deep (max depth is 2, was 3).
┃       lib/crontab/cron_expression/parser.ex:261:13 #(Crontab.CronExpression.Parser.clean_value)

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 0.4 seconds (0.02s to load, 0.4s running checks)
157 mods/funs, found 2 refactoring opportunities.
```

Reply:
OK I've extracting cleaning up of last week day and nth week day into separate functions.